cmd/operator-sdk/generate/bundle/cmd.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bundle
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	kbutil "github.com/operator-framework/operator-sdk/internal/util/kubebuilder"
+)
+
+//nolint:maligned
+type bundleCmd struct {
+	// Options to turn on different parts of bundling.
+	kustomize bool
+	manifests bool
+	metadata  bool
+
+	// Common options.
+	operatorName string
+	version      string
+	inputDir     string
+	outputDir    string
+	manifestRoot string
+	apisDir      string
+	crdsDir      string
+	stdout       bool
+	quiet        bool
+
+	// Metadata options.
+	channels       string
+	defaultChannel string
+	overwrite      bool
+}
+
+//nolint:lll
+func NewCmd() *cobra.Command {
+	c := &bundleCmd{}
+	cmd := &cobra.Command{
+		Use:   "bundle",
+		Short: "Generates bundle data for the operator",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 0 {
+				return fmt.Errorf("command %s doesn't accept any arguments", cmd.CommandPath())
+			}
+
+			// Turn them all on by default here to simplify individual flag usage.
+			fs := cmd.Flags()
+			if !fs.Changed("kustomize") && !fs.Changed("metadata") && !fs.Changed("manifests") {
+				c.kustomize = true
+				c.manifests = true
+				c.metadata = true
+			}
+
+			cfg, err := kbutil.ReadConfig()
+			if err != nil {
+				return fmt.Errorf("error reading configuration: %v", err)
+			}
+			c.setDefaults(cfg)
+
+			if c.kustomize {
+				if err = c.runKustomize(cfg); err != nil {
+					log.Fatalf("Error generating bundle bases: %v", err)
+				}
+			}
+			if c.manifests {
+				if err = c.validateManifests(cfg); err != nil {
+					return fmt.Errorf("invalid command options: %v", err)
+				}
+				if err = c.runManifests(cfg); err != nil {
+					log.Fatalf("Error generating bundle manifests: %v", err)
+				}
+			}
+			if c.metadata {
+				if err = c.runMetadata(); err != nil {
+					log.Fatalf("Error generating bundle metadata: %v", err)
+				}
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().BoolVar(&c.kustomize, "kustomize", false, "Generate kustomize bases")
+	cmd.Flags().BoolVar(&c.manifests, "manifests", false, "Generate bundle manifests")
+	cmd.Flags().BoolVar(&c.metadata, "metadata", false, "Generate bundle metadata and Dockerfile")
+	cmd.Flags().BoolVar(&c.stdout, "stdout", false, "Write bundle manifest to stdout")
```

Comment:
// PS.: we need to clarifies it. Otherwise, is not clear why not just run `make kustomize instead of it`
```suggestion
	cmd.Flags().BoolVar(&c.kustomize, "kustomize", false, "If set, it will generate/update ONLY the kustomize config bases with the specific bundle manifests")
	cmd.Flags().BoolVar(&c.manifests, "manifests", false, "If set, it will generate/update ONLY the bundle manifests directory")
	cmd.Flags().BoolVar(&c.metadata, "metadata", false, "If set, it will generate/update the bundle metadata directory and its Dockerfile")
```